### PR TITLE
Add innovation/lts support for PS starting with ps81

### PR DESCRIPTION
**Pull request by panchal-yash**
Currently for basic review..
Will mod the test_prep.yml once confirmed.

### CODE REVIEW

File: playbooks/ps_lts_innovation.yml [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+  - name: keyring Vault Component and plugins test
+    include_tasks: ../tasks/ps_keyring_vault_component_test.yml
+
+  - name: re-run ps_admin to install rocksdb
+    command: /usr/bin/ps-admin --enable-rocksdb
+
+  - name: install plugins, import world database ps-{{ major_release_version }}
+    command: /package-testing/plugins_test_innovation_lts.sh ps
+
+  - name: check the clone and data masking plugins
```
Comment (eleo007) [inline playbooks/ps_lts_innovation.yml:160]: This plugin checks only clone now. Please rename.

File: playbooks/ps_lts_innovation_major_upgrade_to.yml [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+
+  - name: include tasks for enabling PS 8.0 main repo
+    include: ../tasks/enable_ps8_main_repo.yml
+
+  - name: download and extract world database
+    command: "{{ item }}"
+    with_items:
+    - wget --no-check-certificate -P /package-testing https://raw.githubusercontent.com/Percona-QA/percona-qa/master/sample_db/world.sql
+
+  - name: install Percona Server 8.0 packages
```
Comment (eleo007) [inline playbooks/ps_lts_innovation_major_upgrade_to.yml:62]: Add :
    environment:
      PERCONA_TELEMETRY_DISABLE: "1"

File: tasks/install_pxb_innovation_lts.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+# This task install Percona XtraBackup 8.1 on CentOS and Debian/Ubuntu
```
Comment (eleo007) [inline tasks/install_pxb_innovation_lts.yml:1]: Change version here + when Mohit is back from vacation - coordinate with him (we can commit like this and update it later.)

**Review by eleo007 — APPROVED** [review]

**Review by kaushikpuneet07 — APPROVED** [review]
LGTM , just restore original repo before merge

File: package_check.sh [hunk 4]
```diff
@@ -25,9 +25,9 @@ elif [ $1 = "ps80" ]; then
   release=${PS80_VER#*-}
   revision=${PS80_REV}
 elif [ $1 = "ps81" ]; then
```
Comment (eleo007) [inline package_check.sh:27]: Will it work for ps82?

**Review by kaushikpuneet07 — APPROVED** [review]
Looks good

File: bats/ps-admin_helper.bash [hunk 5]
```diff
@@ -251,7 +251,7 @@ uninstall_all() {
     OPT="--disable-rocksdb"
   fi
   #run bash -c "${PS_ADMIN_BIN} ${CONNECTION} --disable-qrt --disable-audit --disable-pam --disable-pam-compat ${OPT}"
-  if [ ${MYSQL_VERSION} == "8.1" ]; then
+  if [[ "${MYSQL_VERSION}" =~ ^8.[0-9]{1}$ ]]; then
```
Comment (eleo007) [inline bats/ps-admin_helper.bash:254]: Should it match 8.0?

File: package_check.sh [hunk 6]
```diff
@@ -24,10 +24,10 @@ elif [ $1 = "ps80" ]; then
   version=${PS80_VER}
   release=${PS80_VER#*-}
   revision=${PS80_REV}
-elif [ $1 = "ps81" ]; then
-  version=${PS81_VER}
-  release=${PS81_VER#*-}
-  revision=${PS81_REV}  
+elif [ $1 = "ps81" -o $1 = "ps82" ]; then
```
Comment (eleo007) [inline package_check.sh:27]: Are we going to update this script for every release?

File: package_check.sh [hunk 7]
```diff
@@ -86,7 +86,7 @@ product=$1
 log="/tmp/${product}_package_check.log"
 echo -n > $log
 
-if [ ${product} = "ps56" -o ${product} = "ps57" -o ${product} = "ps80" -o ${product} = "ps81" ]; then
+if [ ${product} = "ps56" -o ${product} = "ps57" -o ${product} = "ps80" -o ${product} = "ps81" -o ${product} = "ps82" ]; then
```
Comment (eleo007) [inline package_check.sh:89]: Are we going to update here for every release?

File: package_check.sh [hunk 8]
```diff
@@ -118,7 +118,7 @@ if [ ${product} = "ps56" -o ${product} = "ps57" -o ${product} = "ps80" -o ${prod
         rpm_num_pkgs="8"
         rpm_opt_package="Percona-Server-tokudb-${rpm_maj_version} Percona-Server-rocksdb-${rpm_maj_version}"
       fi
-    elif [ "${product}" = "ps80" -o "${product}" = "ps81" ]; then
+    elif [ "${product}" = "ps80" -o "${product}" = "ps81" -o "${product}" = "ps82" ]; then
```
Comment (eleo007) [inline package_check.sh:121]: Same as ^^

File: package_check.sh [hunk 9]
```diff
@@ -161,7 +161,7 @@ if [ ${product} = "ps56" -o ${product} = "ps57" -o ${product} = "ps80" -o ${prod
       deb_opt_package="percona-server-rocksdb"
       deb_num_pkgs="7"
     fi
-    if [ "${product}" = "ps80" -o "${product}" = "ps81" ]; then
+    if [ "${product}" = "ps80" -o "${product}" = "ps81" -o "${product}" = "ps82" ]; then
```
Comment (eleo007) [inline package_check.sh:164]: Same ^^

File: tasks/test_prep.yml [hunk 10]
```diff
@@ -133,10 +133,10 @@
     with_items:
     - rm -rf /package-testing
     - rm -f master.zip
-    - wget --no-check-certificate -O master.zip https://github.com/Percona-QA/package-testing/archive/master.zip
-    - unzip master.zip
-    - rm -f master.zip
-    - mv package-testing-master /package-testing
+    - wget --no-check-certificate -O POC-INNOVATION-LTS.zip https://github.com/panchal-yash/package-testing/archive/refs/heads/POC-INNOVATION-LTS.zip
```
Comment (eleo007) [inline tasks/test_prep.yml:136]: revert here before commit

File: version_check.sh [hunk 11]
```diff
@@ -22,10 +22,10 @@ elif [ "$1" = "ps80" ]; then
   version=${PS80_VER}
   release=${PS80_VER#*-}
   revision=${PS80_REV}
-elif [ "$1" = "ps81" ]; then
-  version=${PS81_VER}
-  release=${PS81_VER#*-}
-  revision=${PS81_REV}  
+elif [ "$1" = "ps81" -o "$1" = "ps82" ]; then
```
Comment (eleo007) [inline version_check.sh:25]: The same as above - we will have to update the scripts durung every release and we want to overcome this,

File: version_check.sh [hunk 12]
```diff
@@ -54,6 +54,8 @@ elif [ "$1" = "pxb80" ]; then
   version=${PXB80_VER}
 elif [ "$1" = "pxb81" ]; then
   version=${PXB81_VER}
+elif [ "$1" = "pxb82" ]; then
```
Comment (eleo007) [inline version_check.sh:57]: the same as ^^

File: version_check.sh [hunk 13]
```diff
@@ -90,7 +92,7 @@ product=$1
 log="/tmp/${product}_version_check.log"
 echo -n > "${log}"
 
-if [ "${product}" = "ps56" -o "${product}" = "ps57" -o "${product}" = "ps80" -o "${product}" = "ps81" ]; then
+if [ "${product}" = "ps56" -o "${product}" = "ps57" -o "${product}" = "ps80" -o "${product}" = "ps81" -o "${product}" = "ps82" ]; then
```
Comment (eleo007) [inline version_check.sh:95]: The same ^^

File: version_check.sh [hunk 14]
```diff
@@ -114,7 +116,7 @@ if [ "${product}" = "ps56" -o "${product}" = "ps57" -o "${product}" = "ps80" -o
     exit 1
   fi
 
-  if [ ${product} = "ps80" -o ${product} = "ps81" ]; then
+  if [ ${product} = "ps80" -o ${product} = "ps81" -o "${product}" = "ps82" ]; then
```
Comment (eleo007) [inline version_check.sh:119]: The same ^^

File: version_check.sh [hunk 15]
```diff
@@ -187,7 +189,7 @@ elif [ ${product} = "pmm2" -o ${product} = "pmm2-rc" ]; then
   fi
   bash -xe ./check_pmm2_client_upgrade.sh ${version}
 
-elif [ "${product}" = "pxb24" -o "${product}" = "pxb80" -o "${product}" = "pxb81" ]; then
+elif [ "${product}" = "pxb24" -o "${product}" = "pxb80" -o "${product}" = "pxb81" -o "${product}" = "pxb82" ]; then
```
Comment (eleo007) [inline version_check.sh:192]: The same ^^

File: scripts/mysql-comp-test/mysql_comp_test.sh [hunk 16]
```diff
@@ -3,10 +3,10 @@
 set -e
 
 if [ "$#" -ne 1 ]; then
-  echo "This script requires product parameter: ps56, ps57 or ps80 !"
+  echo "This script requires product parameter: ps56, ps57 or ps8x !"
   echo "Usage: ./comp_test.sh <prod>"
   exit 1
-elif [ $1 != "ps56" -a $1 != "ps57" -a $1 != "ps80" -a $1 != "ps81" ]; then
+elif ! [[ "$1" =~ ^ps5[0-9]{1}$ || "$1" =~ ^ps8[0-9]{1}$i ]]; then
```
Comment (eleo007) [inline scripts/mysql-comp-test/mysql_comp_test.sh:9]: We can have only 57 or 56, no need for regexp here

**Review by eleo007 — APPROVED** [review]

**Review by kaushikpuneet07 — APPROVED** [review]
LGTM

**Review by mohitj1988 — APPROVED** [review]

**Pull request merged**